### ISSUE TITLE
Premissive mode

### ISSUE THREAD

**Issue by lonix** [hadolint/hadolint#74]
Posible to add a permissive mode, as in Prints the error but exit 0 anyway, usefull for CI intigration.

**Comment by lukasmartinelli** [comment]
> Posible to add a permissive mode, as in Prints the error but exit 0 anyway, usefull for CI intigration.

In such cases I usually execute something like `hadolint || true` - if you have access to bash.
Does that help?

**Comment by lonix** [comment]
I Do use that, but i consider that a workaround.

one could also consider haveing a score based system, before exit 1.

**Comment by lukasmartinelli** [comment]
> I Do use that, but i consider that a workaround.
> 
> one could also consider haveing a score based system, before exit 1.
> 
> e.g. allow 2 errors of "low category"

So actually it is more about allowing certain errors - not the exit code.
You can exclude rules.

```
hadolint - Dockerfile Linter written in Haskell

Usage: hadolint [-v|--version] [--ignore RULECODE] [DOCKERFILE...]
  Lint Dockerfile for errors and best practices

Available options:
  -h,--help                Show this help text
  -v,--version             Show version
  --ignore RULECODE        Ignore rule
```

How do UNIX programs usually handle that and what are their standard flags for that?
`-fpermissive` ?

**Comment by aw-was-here** [comment]
> You can exclude rules.

It would be better to be able to exclude rules on a per-line basis, similar to how shellcheck does it.

**Comment by lorenzo** [comment]
@aw-was-here how would that work?

**Comment by aw-was-here** [comment]
Most lint systems have a way to specify a "temporary" rule exclusion.  Shellcheck in particular allows for a comment in the line prior to a code block. e.g.:

```bash
# shellcheck disable=SC2034
IFS=$'\n' sa=($(sort <<<"${array[*]}"))
```

when run through shellcheck, the IFS line wil not throw a [SC2034](https://github.com/koalaman/shellcheck/wiki/SC2034) error

Similarly, I could see hadolint doing something like:

```
# hadolint disable=DL3008,DL3015
RUN apt-get install foo
```

**Comment by lorenzo** [comment]
Interesting, that makes sense to me. Thanks for clarifying @aw-was-here

**Comment by lorenzo** [comment]
I'll close this ticket an open a new one with that idea
